src/main/java/org/spongepowered/api/item/inventory/InventoryTransformations.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import org.spongepowered.api.item.inventory.entity.Hotbar;
+import org.spongepowered.api.item.inventory.entity.MainPlayerInventory;
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+
+public class InventoryTransformations {
+
+    public static final InventoryTransformation NONE = DummyObjectProvider.createFor(InventoryTransformation.class, "NONE");
+
+    /**
+     * A Transformation that puts the Hotbar of a player inventory before the main inventory.
+     * <p>
+     *     This transformation is equivalent to {@link Inventory#union(Inventory)} together a query
+     *     for {@link Hotbar} and {@link MainPlayerInventory} in this order
+     * </p>
+     */
+    public static final InventoryTransformation HOTBAR_FIRST = DummyObjectProvider.createFor(InventoryTransformation.class, "HOTBAR_FIRST");
```

Comment:
The example is wrong in this case, since the `MainPlayerInventory` contains the main grid (3 rows) and the hotbar.

And transforming a `PlayerInventory` and `MainPlayerInventory` should give a different result. Transforming `MainPlayerInventory` would just include the main grid and hotbar where transforming the `PlayerInventory` would still include the equipment.

Reply:
In lantern I am using a `View` which gives all the different kind of player inventory views used in vanilla minecraft:
https://github.com/LanternPowered/LanternServer/blob/master/src/main/java/org/lanternpowered/server/inventory/vanilla/AbstractUserInventory.java#L164

Reply:
How is it wrong?
This is basicly your MAIN_AND_PRIORITY_HOTBAR View.
It filters out the EquipmentSlots as well as the Crafting slots in the Player Container and orders the hotbar before the grid.

Adding more predefined Transformations is possible.
I just thought this one would be the most used one.

e.g. Hotbar + Grid + Equipment would be:
`InventoryTransformation.of(INVENTORY_TYPE.of(Hotbar.class), INVENTORY_TYPE.of(MainPlayerInventory.class),
INVENTORY_TYPE.of(EquipmentInventory.class))`

Reply:
You're right, the doc is good the way it is, I was confused because the `MainPlayerInventory` already includes the `Hotbar`.

But I still think it would make more sense to allow `HOTBAR_FIRST` to be target specific, for example:
```
PlayerInventory inventory = ...;
// The following gives:
// InventoryTransformation.of(INVENTORY_TYPE.of(Hotbar.class), INVENTORY_TYPE.of(MainPlayerInventory.class), INVENTORY_TYPE.of(EquipmentInventory.class))
Inventory prioritizedHotbar = inventory.transform(InventoryTransformations.HOTBAR_FIRST);
// The following gives:
// InventoryTransformation.of(INVENTORY_TYPE.of(Hotbar.class), INVENTORY_TYPE.of(MainPlayerInventory.class))
inventory = inventory.getMain().transform(InventoryTransformations.HOTBAR_FIRST);
// 
```
Or if you prefer not to, change the transformation field name to say that it's only the main inventory. (to avoid confusion and conflicts with possible other transformations)

Reply:
`Inventory#union` operates on the Set of Slots so no duplicates.

What do you mean by target specific?
Applying `InventoryTransformation.of(INVENTORY_TYPE.of(Hotbar.class), INVENTORY_TYPE.of(MainPlayerInventory.class), INVENTORY_TYPE.of(EquipmentInventory.class))`
to `inventory.getMain()` would result in the same result (as without EquipmentInventory) because the `#getMain` does not contain an `EquipmentInventory`.

The rename would be fine.